doc/adr/0011-tsp-analysis-api.md

```diff
@@ -178,100 +183,175 @@ sequenceDiagram
+
+
+Example configuration source type:
+
+ConfigurationSourceType {
+    "name": Custom Flame Chart,
+    "description": Create instance of a custom flame based on input parameters,
+    "id": "custom.flame.charts.id",
+    "scope": "data provider",
+    "parameterDescriptors": [
+        {
+            "keyName": "title",
+            "description": "Provide a name of custom flame chart to be shown in UI",
+            "dataType": "string",
+            "isRequired": "true"
+        },
+        {
+            "keyName": "filter",
+            "description": "Provide a filter string according to filter language",
+            "dataType": "string",
+            "isRequired": "true"
+        }
+    ]
+}
+```
+
+Input data provider:
+
+```javascript
+OutputDescriptor {
+    "id": "flamechart.id",
+    "name": "Flame Chart",
+    "description": "Flame Chart description",
+    "type": "TIME_GRAPH"
+}
 ```
 
-#### Sequence: Delete configuration instance for an experiment
+`User input`:
+```javascript
+    {"title": "CPU 0-1 only", "filter": "cpu matches [0-1]"}
+```
+
+Resulting data provider:
+```javascript
+    OutputDescriptor {
+        id: string,
+        name: string,
+        description: string,
+        type: string,  // provider type
+        configurtaton: Configuration
+    }
+```
+
+Notes:
+- Stored configurations need to have unique IDs. Consider generating IDs from input parameters to avoid duplicate configurations.
+
+#### Sequence: Delete custom data provider
 
-The following illustrates the sequence of events and messages to delete an configuration instance for a given type and experiment.
+The following illustrates the sequence of events and messages to delete a custom data provider for an experiment.
 
-Pre-requisite: Analysis instance created as described in [Sequence: Create configuration instance for an experiment](#sequence-create-configuration-instance-for-an-experiment).
+Pre-requisite: Analysis instance created as described in [Sequence: create new data provider](#sequence-create-new-data-provider).
 
 ```mermaid
 sequenceDiagram
     participant user as User
     participant client as TSP Client
     participant server as Trace Server
-    user->>client: Open Configuration Selector UI for experiments
-    client->>server: GET /experiments/{expUUID}/outputs/config/
-    server->>client: 200: Map <typeId -> List of Configuration>
-    client->>client: Populate drop-down menu
-    user->>client: Select "In-And-Out" type
-    client->>client: Populate UI with "In-And-Out" type only
-    user->>client: Select configuration instance
-    user->>client: CLick on Delete button
-    client->>server: DELETE /config/types/{typeId}/configs/{configId}
-    server->>client: 200
-    client->>client: Refresh UI
+    client->>server: GET /experiments/{expUUID}/outputs
+    server->>client: 200: List of Available Views
+    client->>client: Render "Available Views" view and render Delete button for derived outputs
```

Comment:
How do we know that it's a derived output? Is it because of the presence of parentId? Or configuration? Or both?

Reply:
The presence of a configuration should be used to determine that's a derived output. ParentId might not be sufficient in case it's used for ways of grouping.